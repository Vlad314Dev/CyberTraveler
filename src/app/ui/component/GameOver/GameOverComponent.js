import './GameOverStyle';

import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

class GameOverComponent extends PureComponent
{
    static propTypes = {
        score: PropTypes.number.isRequired
    }

    render()
    {
        const {
            score
        } = this.props;

        return (
            <div block="GameOver">
                <div block="GameOver" elem="Main">
                    <h1>Congratulations, you have completed the game!</h1>
                    <h2>{`Your score: ${ score }`}</h2>
                </div>
                <div block="GameOver" elem="CreditsWrapper">
                    <h1>Credits</h1>
                    <div block="GameOver" elem="Credits">
                        <div block="GameOver" elem="Section">
                            <h2>Art</h2>
                            <p>Luis Zuno (www.ansimuz.com)</p>
                        </div>
                        <div block="GameOver" elem="Section">
                            <h2>Music</h2>
                            <p>Luis Zuno (www.ansimuz.com)</p>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

export default GameOverComponent;
