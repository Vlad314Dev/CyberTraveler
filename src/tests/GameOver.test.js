/* eslint-disable no-undef */
import { shallow } from 'enzyme';
import React from 'react';

import GameOverComponent from '../app/ui/component/GameOver/GameOverComponent';

describe('GameOver', () => {
    it('Should render', () => {
        // eslint-disable-next-line no-unused-vars
        const component = shallow(
            <GameOverComponent 
                score={ 0 }
            />
        );
    });

    it('Should render layout', () => {
        const component = shallow(
            <GameOverComponent 
                score={ 0 }
            />
        );

        expect(component.getElements()).toMatchSnapshot();
    });
});
