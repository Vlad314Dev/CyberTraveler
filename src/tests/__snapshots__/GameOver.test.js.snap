// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameOver Should render layout 1`] = `
Array [
  <div
    className="GameOver"
  >
    <div
      className="GameOver__Main"
    >
      <h1>
        Congratulations, you have completed the game!
      </h1>
      <h2>
        Your score: 0
      </h2>
    </div>
    <div
      className="GameOver__CreditsWrapper"
    >
      <h1>
        Credits
      </h1>
      <div
        className="GameOver__Credits"
      >
        <div
          className="GameOver__Section"
        >
          <h2>
            Art
          </h2>
          <p>
            Luis Zuno (www.ansimuz.com)
          </p>
        </div>
        <div
          className="GameOver__Section"
        >
          <h2>
            Music & SFX
          </h2>
          <p>
            Mysic by Karl Casey @ White Bat Audio
          </p>
          <p>
            Sound from Zapsplat.com
          </p>
        </div>
      </div>
    </div>
  </div>,
]
`;
